tijanabekic/assignment2/Assignment 2.java

```diff
@@ -0,0 +1,126 @@
+//Q1. Given a binary tree and a key, write a function that prints all the ancestors of the key in the given binary tree.
+
+public class Node {
+	int data;
+	Node left;
+	Node right;
+	Node parent;
+
+	Node (int d, Node par) {
+		data = d;
+		parent = par;
+	}
+}
+
+public class Q1 {
+	public static void printAllAncestors(Node node) {

```

Comment:
If you change this method to return a List of the ancestor nodes, then your tests can check that
printAllAncestors matches the expected result without resorting to a user reading the program's output. When there are lots of tests it is easier to only draw attention to the failing ones.